src/en/kickassanime/src/eu/kanade/tachiyomi/animeextension/en/kickassanime/KickAssAnime.kt

```diff
@@ -0,0 +1,322 @@
+
+        sources.forEach { source ->
+            when (source.jsonObject["name"]!!.jsonPrimitive.content) {
+                "BETAPLAYER" -> {
+                    videoList.addAll(
+                        extractBetaVideo(
+                            source.jsonObject["src"]!!.jsonPrimitive.content,
+                            source.jsonObject["name"]!!.jsonPrimitive.content
+                        )
+                    )
+                }
+                "BETASERVER3" -> {}
+                else -> {
+                    videoList.addAll(
+                        extractVideo(
+                            source.jsonObject["src"]!!.jsonPrimitive.content,
+                            source.jsonObject["name"]!!.jsonPrimitive.content
+                        )
+                    )
+                }
+            }
+        }
+        return videoList
+    }
+
+    private fun extractVideo(serverLink: String, server: String): List<Video> {
+        val playlistInterceptor = MasterPlaylistInterceptor()
+        val kickAssClient = client.newBuilder().addInterceptor(playlistInterceptor).build()
+        kickAssClient.newCall(GET(serverLink)).execute()
+        val data = playlistInterceptor.playlist
+        val playlist = mutableListOf<Video>()
+        val subsList = mutableListOf<Track>()
+
+        if (server == "MAVERICKKI") {
+            val subLink = serverLink.replace("embed", "api/source")
+            val subResponse = Jsoup.connect(subLink).ignoreContentType(true).execute().body()
+            val json = Json.decodeFromString<JsonObject>(subResponse)
+            json["subtitles"]!!.jsonArray.forEach {
+                val subLang = it.jsonObject["name"]!!.jsonPrimitive.content
+                val uri = Uri.parse(serverLink)
+                val subUrl = "${uri.scheme}://${uri.host}" + it.jsonObject["src"]!!.jsonPrimitive.content
+                try {
+                    subsList.add(Track(subUrl, subLang))
+                } catch (e: Error) {}
+            }
+        }
+
+        data.forEach { playlistPair ->
+            val (videoLink, headers) = playlistPair
+            val masterPlaylist = client.newCall(GET(videoLink, headers)).execute().body!!.string()
+            masterPlaylist.substringAfter("#EXT-X-STREAM-INF:")
+                .split("#EXT-X-STREAM-INF:").map {
+                    val quality = it.substringAfter("RESOLUTION=").split(",")[0].split("\n")[0].substringAfter("x") + "p $server" +
+                        if (subsList.size > 0) { " (Toggleable Sub Available)" } else { "" }
+                    var videoUrl = it.substringAfter("\n").substringBefore("\n")
+                    if (videoUrl.startsWith("https").not()) {
+                        val pos = videoLink.lastIndexOf('/') + 1
+                        videoUrl = videoLink.substring(0, pos) + videoUrl
+                    }
+                    playlist.add(Video(videoUrl, quality, videoUrl, subtitleTracks = subsList, headers = headers))
```

Comment:
```suggestion
                    try {
                        playlist.add(Video(videoUrl, quality, videoUrl, subtitleTracks = subsList, headers = headers))
                    } catch (e: Error) {
                        playlist.add(Video(videoUrl, quality, videoUrl, headers = headers))
                    }
```

I think we need to do this because it would crash on the current stable release.